packages/components/Tabs/Tabs.html

```diff
@@ -0,0 +1,118 @@
+<div class="tabs">
+  {#if _tabs.length}
+  <table class="tab-navigation">
+    <tr>
+      {#each _tabs as tab, key}
+      <td on:click="set({index: key})" class:active="key === index">
+        {tab.title}
+        <div class="line"></div>
+      </td>
+      {/each}
+    </tr>
+  </table>
+
+  {/if}
+
+  <div class="tabs-paine-container" ref:container>
+    <slot></slot>
+  </div>
+</div>
+
+<script>
+  export default {
+    data() {
+      return {
+        _tabs: [],
+        _lastIndex: 0,
+        index: 0,
+      };
+    },
+    oncreate() {
+      const _tabs = [].filter.call(
+        this.refs.container.children,
+        element =>
+          element.classList.contains('tab-pane') &&
+          element.title !== 'undefined',
+      );
+
+      this.set({ _tabs });
+    },
+    onupdate({ changed, current }) {
+      if (changed.index === true || changed._tabs === true) {
+        // Update tab position
+        this._changeTab(current.index);
+      }
+    },
+    methods: {
+      _validIndex(index) {
+        const { _tabs } = this.get();
+        switch (true) {
+          case index < 0:
+            return 0;
+          case index >= _tabs.length:
+            return _tabs.length - 1;
+          default:
+            return index;
+        }
+      },
+      _changeTab(index) {
+        const { _tabs, _lastIndex: lastIndex } = this.get();
+        if (_tabs.length) {
+          // Prevent redraw if index is the same of lastIndex
+          if (lastIndex !== index) {
```

Comment:
Se for a mesma tab, o evento de 'change' deve disparar? Talvez se inverter essa lógica e dar um 'return' logo de cara resolva isso

Reply:
Com o onupdate agora, essa lógica do if acho que não tem mais sentido.
Era pra blocar se a aba atual fosse clicada de novo, ele iria esconder-mostrar a mesma aba.

Reply:
tirei o if